usr/sbin/p11sak/p11sak.c

```diff
@@ -2566,11 +2567,17 @@ int main(int argc, char *argv[])
     if (!pin) {
         printf("Please enter user PIN:");
         rc = get_pin(&pin, &pinlen);
+        pin_alloced = ckb_true;
 
-        if (strlen(pin) == 0) {
+        if (pin == NULL || strlen(pin) == 0) {
             char *s = getenv("PKCS11_USER_PIN");
             if (s) {
-                strcpy((char*) pin, s);
+                free(pin);
+                pin = strdup(s);
+                if (!pin) {
+                    rc = -ENOMEM;
```

Comment:
Use CKR_HOST_MEMORY ?

Reply:
Yea, get_pin does not return CKR_ errors but -1 or negative errnos, so that lead me to use negative errnos. But its certainly better to use CKR_ errors here. Will fix this.